pkg/controller/constrainttemplate/constrainttemplate_controller_test.go

```diff
@@ -648,35 +623,27 @@ func applyCRD(ctx context.Context, g *gomega.GomegaWithT, client client.Client,
 
 func deleteObject(ctx context.Context, c client.Client, obj client.Object, timeout time.Duration) error {
```

Comment:
It's now a bit confusing to have this `deleteObject()` and `testutils.DeleteObject`.  

It looks like this one just deletes and adds on a polling mechanism to confirm deletion before moving on. 

Could we rename this function to reflect this?  `deleteObjectAndConfirm`?

It could also use `testutils.DeleteObject` internally, reflecting that its core task is the same.